worlds/sc2/regions.py

```diff
@@ -288,9 +300,9 @@ def remove_missions(world: 'SC2World', mission_order: Dict[str, Dict[str, Any]],
             this_layout = f"Mission Pack {idx + 2}"
             prev_layout = f"Mission Pack {idx + 1}"
             prev_last_index = mission_order[SC2Campaign.NCO.campaign_name][prev_layout]["size"] - 1
-            mission_order[SC2Campaign.NCO.campaign_name][this_layout]["entry_rules"] = [{
+            mission_order[SC2Campaign.NCO.campaign_name][this_layout]["entry_rules"][0] = {
```

Comment:
Is it possible for a user to specify additional entry rules on a preset? If so, this would override it. I'd also just expect a `.append()` rather than always writing to index 0 if it already exists.

Also asking for a sanity check that the list is already constructed. The old version creates and assigns a list, the new code modifies a list, and I don't see any change that creates the list. Object creation is many functions away and difficult to look up and the type has an `Any` meaning it's not easy to tell if this list is guaranteed to be created already.

Reply:
This is pretty far out of context, but surrounding code creates several guarantees.
1. This is specifically a non-custom vanilla shuffled mission order, because that's the only place `remove_missions()` is called, so a user can't specify additional rules for the mission order this is working with.
2. The list is guaranteed to be constructed because this mission order has gone through `CustomMissionOrder(mission_order).value` before the `remove_missions()` call and NCO must be enabled, so the preset is loaded and its options are set up.
This means we have the guarantee that `entry_rules` here matches `entry_rules` in the NCO preset layouts, which previously meant that overwriting the old entry rule list was fine, but now that there's a second entry rule for key logic it needs to target the first rule directly.

...Or *would* need to, if this wasn't code I forgot to clean up with this PR. Since I've enabled the titles in the client for the NCO Mission Packs these entry rules have changed format and don't need this special-casing anymore. I'm removing this section of code.

Regarding sanity checking more generally, there's a billion places where it would make some sense to have sanity checks, which I think is sort of unavoidable when we're necessarily working with a YAML-based dict as the base structure. Instead, one of the responsibilities of the `CustomMissionOrder` option class is to enforce its schema, so anything that has gone through the option can make strong assumptions about what is or isn't present in the mission order dict. I know it's not a *great* solution, but IMO it makes working with mission orders in code a lot more reasonable without sacrificing readability.